### PR TITLE
Fix config file parsing

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Remove `serde-hex` since it's no longer compatible with the new `toml` version
- Include the `hex` crate
- Add functions for serializing and deserializing
- Add test for serialization and deserialization methods

Not sure if this is the best way, but the file needed to be backward compatible.

### CODE REVIEW

**Review by MabezDev — CHANGES_REQUESTED** [review]

File: espflash/src/cli/config.rs [hunk 1]
```diff
@@ -94,3 +120,81 @@ impl Config {
...
+    struct TestData {
+        #[serde(serialize_with = "parse_u16_hex", deserialize_with = "parse_hex_u16")]
+        value: u16,
+    }
+
+    #[test]
+    fn test_parse_hex_u16() {
+        // Test no padding
+        let input = "aaaa";
+        let result: Result<TestData, _> = toml::from_str(&format!("value = \"{}\"", input));
```
Comment (bjoernQ) [inline espflash/src/cli/config.rs:139]: probably a matter of taste but I would write something less verbose like
`let result: Result<TestData, _> = toml::from_str(r#"value = "aaaa""#);`

but really a matter of taste

**Comment by bjoernQ** [comment]
maybe you also want to adapt `test_parse_u16_hex`

**Comment by SergioGasquez** [comment]
> maybe you also want to adapt `test_parse_u16_hex`

Ups, forgot to push the changes after updating it. Thanks!

**Review by bjoernQ — APPROVED** [review]
LGTM

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

**Pull request merged**